### ISSUE TITLE
Weird behavior when two issues have the same name

### ISSUE THREAD

**Issue by yoyozbi** [benr77/obsidian-gitlab-issues#30]
| Question                           | Answer        |
| ---------------------------------- | ----------------- |
| **Obsidian version**    |     v1.7.4        |
| **Plugin version**         |   0.1.10         |
| **Operating System**  |     NixOS 24.05  |

**Describe the bug**
I have a repository where I have multiple issues with the same name. I know it could seem weird to do that, but I separate them using tags. When doing so, only the first issue file is created 

**To Reproduce**
Create two issues with the same title.

**Expected behavior**
Without changing the current default behavior, you could add a way to specify a template for filenames. Or change the current behavior to have a prefix with the issue ID in the filename

**Additional context**
Feel free to ask me if you don't have time, I could try to do it.

**Comment by benr77** [comment]
Yes, this will be the case because the plugin writes a file for each issue, and the filename is derived from the issue title.

I prefer your solution of using the issue ID as a prefix. However, this would be a BC break so I guess we'd need to make this configurable and ensure the default behaviour remains unchanged (i.e. not using the ID as a prefix). 

Adding a config setting for a "filename template", which is a string that is parsed to substitute variables such as "id", "title" etc. Then the default setting for "filename template" could be just `{slug}.md` but you would have the option to change this to `{id}-{slug}.md` etc.

I have very little time available at the moment, and this is definitely more of an edge case, so doubt I will work on it, but a PR from you or anybody else is more than welcome. If you do submit a PR please ensure it has some test cases included as the project is lacking test coverage. Thanks.
